L1Trigger/TrackFindingTracklet/test/L1TrackNtupleMaker.cc

```diff
@@ -998,32 +1003,64 @@ void L1TrackNtupleMaker::analyze(const edm::Event& iEvent, const edm::EventSetup
         else
           myFake = 1;
 
-        myTP_pdgid = my_tp->pdgId();
-        myTP_pt = my_tp->p4().pt();
-        myTP_eta = my_tp->p4().eta();
-        myTP_phi = my_tp->p4().phi();
-        myTP_z0 = my_tp->vertex().z();
+        tmp_matchtp_pdgid = my_tp->pdgId();
+        tmp_matchtp_pt = my_tp->pt();
+        tmp_matchtp_eta = my_tp->eta();
+        tmp_matchtp_phi = my_tp->phi();
 
-        float myTP_x0 = my_tp->vertex().x();
-        float myTP_y0 = my_tp->vertex().y();
-        myTP_dxy = sqrt(myTP_x0 * myTP_x0 + myTP_y0 * myTP_y0);
+	float tmp_matchtp_vz = my_tp->vz();
+	float tmp_matchtp_vx = my_tp->vx();
+	float tmp_matchtp_vy = my_tp->vy();
+        tmp_matchtp_dxy = sqrt(tmp_matchtp_vx * tmp_matchtp_vx + tmp_matchtp_vy * tmp_matchtp_vy);
+
+	// ----------------------------------------------------------------------------------------------
+	// get d0/z0 propagated back to the IP
+
+	float tmp_matchtp_t = tan(2.0 * atan(1.0) - 2.0 * atan(exp(-tmp_matchtp_eta)));
+
+	float delx = -tmp_matchtp_vx;
```

Comment:
I think the minus signs on L1021 and L1022 might be wrong.

Reply:
I don't know enough about propagating back to know if this is wrong or not. If you can confirm that this is incorrect then I will make the change. Do you also think that the overall sign of d0 is wrong? Ian proposes that it might be incorrect.

Reply:
To your second point on whether or not the overall sign of d0 is wrong, I agree with Ian that the sign of d0 does not seem to follow the cms convention. The easiest way to see this is to plot tp_d0 vs tp_d0_prod 
![d0vsd0_prod_default](https://user-images.githubusercontent.com/43122048/114689467-0396fc00-9d16-11eb-99de-fcd149d9ef5d.png) 
There seems to be a negative association between the two variables and I am pretty sure the sign of tp_d0_prod is correct. 
The only way I can think of to fix this sign problem is NOT directly flipping the sign on L1163. Instead, I believe the signs on L1151&L1152 are incorrect. If we modify those two lines and plot tp_d0 vs tp_d0_prod again:
![d0vsd0_prod_fixsign](https://user-images.githubusercontent.com/43122048/114690637-2ece1b00-9d17-11eb-936f-fc41a6417fa0.png).
Let me talk to Louise to see we can reach an agreement.